server/api/utils/clients/async_nuclio.py

```diff
@@ -43,8 +46,33 @@ async def list_api_gateways(self, project_name=None):
         return await self._send_request_to_api(
             method="GET",
             url=self._nuclio_dashboard_url,
+            path=NUCLIO_API_GATEWAYS_ENDPOINT.format(api_gateway=""),
+            headers=headers,
+        )
+
+    async def create_api_gateway(
+        self,
+        project_name: str,
+        api_gateway_name: str,
+        api_gateway: mlrun.common.schemas.APIGateway,
+    ):
+        headers = {}
+
+        if project_name:
+            headers[NUCLIO_PROJECT_NAME_HEADER] = project_name
+
+        body = self._generate_nuclio_api_gateway_body(
+            project_name=project_name,
+            api_gateway_name=api_gateway_name,
+            api_gateway=api_gateway,
+        )
+
+        return await self._send_request_to_api(
+            method="PUT",
```

Comment:
Will it create a new API gateway even if it does NOT exist?

Reply:
@TomerShor NO, it won't. Because inside of nuclio we do `get` before the resource update. Maybe we can consider moving Update to CreateOrUpdate logic in Nuclio, but for now will leave everything on `POST`

Reply:
We can implement the Create or Update logic inside mlrun.
Let's leave this PR with `create` only, so we can move on :)